interfaces/apparmor/spec.go

```diff
@@ -477,3 +491,21 @@ func (spec *Specification) AddPermanentSlot(iface interfaces.Interface, slot *sn
 	}
 	return nil
 }
+
+// UsesPtraceTrace records when to omit explicit ptrace deny rules
+func (spec *Specification) UsesPtraceTrace() {
```

Comment:
I would have expected `UsesPtraceTrace` to return `bool` and not set any internal state. How about calling the set `UsePtraceTrace`, `SupressPtraceTrace`. Since this is all a part of the package we could drop the getter functions entirely and just either expose a single public function that computes `uses && !suppress` or just access the fields directly.

Reply:
I agree with zyga, this is a bit unusual in terms of go style. Most getters of the form "GetFoo() bool" are simply called "Foo() bool". Maybe something along the lines of what zyga suggested, e.g. http://paste.ubuntu.com/p/chSKC2jSBp/ ? This diff would add an explicit Set{Uses,Suppresses}PtraceTrace which simplifies some if/else. I kept the Get* to keep the diff small but I would like to rename them as zyga suggested. Happy to work on this fwiw.

Reply:
I'm fine with reorganizing as requested. I'd prefer not to have Set* allow setting both ways because the interfaces will only need to call it with `true`. If they call it with anything else, then it could introduce difficult to diagnose bugs where sometimes the rule is added and sometimes not based on on interface connection ordering. Eg, interface foo uses `SetSuppressPtraceTrace(true)`, then interface bar does `SetSuppressPtraceTrace(false)`. We'll get different behavior depending on which one snapd decides to connect first. Unless you strongly feel otherwise, I'll leave the one-way behavior (but still name it Set*).

Reply:
Done

Reply:
@zyga - regarding dropping getter, I use them in common_test.go and felt this was cleaner.

Reply:
I noticed but for such cases we can use `export_test.go`, though at this stage it's not a strong feeling.

Reply:
@zyga - I recall having some trouble with mocking certain things in the PRs though I can't recall if this was one of them. If you don't have a strong feeling that it is a blocker, I suggest letting it in as is (it might be useful down the line). If it is strong enough for you wanting me to do a followup PR, I can.